### ISSUE TITLE
Keyboard shortcuts don't do anything

### ISSUE THREAD

**Issue by dannyharding10** [onivim/oni2#186]
The normal text editor keyboard shortcuts don't do anything. cmd-O should let me browse for a project I want to open. cmd-S should attempt to save, and in an unnamed document it should let me pick a name and a location.

**Comment by Akin909** [comment]
@dannyharding10 we're pre-alpha at the moment, so there's still *a lot* to do but when we get closer to alpha we should hopefully have fleshed things out more

**Comment by dannyharding10** [comment]
@Akin909 Yes makes sense! I spoke briefly with @bryphe about how I could help out on the project. He said there aren’t many issues being tracked, but that I could help by opening up the project, finding issues, and reporting them as Issues. I know that this one and the other one that I logged are pretty obvious, but they are the first ones I found! If you have any tips on better ways to jump in and help, I want to know! I would love to help out

**Comment by jrwrigh** [comment]
Shouldn't something like saving a file be left as the standard neovim default? `:w`?

'Open project' could probably be left as a ex command and then binded by the user if they'd like.

**Comment by dannyharding10** [comment]
@jrwrigh I was under the impression that onivim was going to be an option for non-vim users as well, which would require it to include some standard functional like this. @bryphe will this project target more than vim users?

**Comment by bryphe** [comment]
> @jrwrigh I was under the impression that onivim was going to be an option for non-vim users as well, 

We definitely would like to get there!

It's a tough balance to provide an out-of-box experience that caters to _both_ Vim and non-Vim users - but my plan at the moment is to provide the 'vim-style' keybindings out of the box, but support VSCode-style keybindings as a plugin (ie, the _inverse_ of `VSCodeVim`- like `OniVsCode`). We could potentially expose this as an option on the Welcome screen so it's easier to toggle / switch.

A few pieces that we know are challenging:
- `<C-n>` - overloaded in insert mode (starts completion, moves to next item) in Vim, but would be 'new file' in VSCode.
- `<C-o>` - overloaded in normal mode, but would be 'open file' in VSCode.

Also, a key differentiator would be whether you _start_ in normal mode vs insert mode. There's also some other subtler things that we'll have to think through in the UX (visual selection vs 'selection' mode - if you select text with a mouse, which mode should it be?)

**Comment by dannyharding10** [comment]
Regardless, I am very excited for this project! I’m looking forward to start alpha and beta testing it

**Comment by Spirarel** [comment]
> > @jrwrigh I was under the impression that onivim was going to be an option for non-vim users as well,
> 
> We definitely would like to get there!
> 
> It's a tough balance to provide an out-of-box experience that caters to _both_ Vim and non-Vim users - but my plan at the moment is to provide the 'vim-style' keybindings out of the box, but support VSCode-style keybindings as a plugin (ie, the _inverse_ of `VSCodeVim`- like `OniVsCode`). We could potentially expose this as an option on the Welcome screen so it's easier to toggle / switch.
> 
> A few pieces that we know are challenging:
> 
> * `<C-n>` - overloaded in insert mode (starts completion, moves to next item) in Vim, but would be 'new file' in VSCode.
> * `<C-o>` - overloaded in normal mode, but would be 'open file' in VSCode.
> 
> Also, a key differentiator would be whether you _start_ in normal mode vs insert mode. There's also some other subtler things that we'll have to think through in the UX (visual selection vs 'selection' mode - if you select text with a mouse, which mode should it be?)

I think those are actually done with cmd+n and cmd+o on VSC. It could be my keymap, but I though this was the case. Since the terminal captures the `cmd` modifier, it's actually nice to use these chords for not-vim, VSC-like features since the mapping is a superset of what vimmers expect.
